src/get-started/light-client.md

```diff
@@ -356,50 +474,71 @@ Otherwise, transactions cannot be verified against the proof:
 
-Next, we validate each transaction signature with `Transaction.verifySignature`
-method.
+- calculate a hash from the transaction buffer with the `Exonum.hash`
+  method. Compare it with the corresponding hash from the array of transaction
+  hashes in the proof
+- serialize the transaction in its explicit view and compare it with the
+  serialized version of the transaction body (buffer without signature). The
+  explicit view of the transaction is stored in the `debug` field thereof, while
+  the serialized view thereof is stored in the `message` field in the hex format
+- validate each transaction signature with the `Transaction.verifySignature`
+  method. The required data for validation are signature and author of the
+  transaction in the hex format and the transaction data in its explicit view.
 
-Finally, we calculate a hash from a transaction body with `Transaction.hash`
-method to compare it with the corresponding hash from the proof.
+Below is an example of the proof check for the transaction history of a wallet:
 
 ```javascript
-data.wallet_history.transactions.forEach((transaction, index) => {
-  // generate transaction definition
-  const Transaction =  new Exonum.newMessage({
-    protocol_version: 0,
-    service_id: 128,
-    message_id: 2,
-    fields: [
-      { name: 'from', type: Exonum.PublicKey },
-      { name: 'to', type: Exonum.PublicKey },
-      { name: 'amount', type: Exonum.Uint64 },
-      { name: 'seed', type: Exonum.Uint64 }
-    ],
-    signature: transaction.signature
+for (let transaction of data.wallet_history.transactions) {
+  const hash = transactionsMetaData[index++]
+  const buffer = Exonum.hexadecimalToUint8Array(transaction.message)
+  const bufferWithoutSignature = buffer.subarray(0, buffer.length - 64)
+  const author = Exonum.uint8ArrayToHexadecimal(buffer.subarray(0, 32))
+  const signature = Exonum.uint8ArrayToHexadecimal(
+    buffer.subarray(buffer.length - 64, buffer.length));
+
+  const Transaction = new Exonum.newTransaction({
+    author: author,
+    service_id: SERVICE_ID,
+    message_id: TX_WALLET_ID,
+    schema: proto.exonum.examples.cryptocurrency_advanced.Issue
   })
 
-  // validate transaction signature
-  if (!Transaction.verifySignature(transaction.signature,
-    transaction.body.from, transaction.body)) {
-    throw new Error('Invalid transaction signature has been found')
+  // Calculate a hash from the transaction body and compare it with the
+  // corresponding hash in the array of transaction hashes.
+  if (Exonum.hash(buffer) !== hash) {
+     throw new Error('Invalid transaction hash')
   }
 
-  // validate hash
-  if (Transaction.hash(transaction.body) !== transactionsMetaData[index]) {
-    throw new Error('Invalid transaction hash has been found')
+  // Serialize transaction from the debug view and compare it with the
+  // `bufferWithoutSignature` obtained above.
+  if (!Transaction.serialize(transaction.debug).every(function (el, i) {
```

Comment:
Nit: wouldn't it make more sense to use an arrow function syntax here? I don't think it's supported less than `for .. of`.

Reply:
Make sense, done